source/GameInterface/Services/Villages/Patches/ChangeVillageStatePatch.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Common;
+using Common.Messaging;
+using Common.Util;
+using GameInterface.Services.Villages.Messages;
+using HarmonyLib;
+using System;
+using System.Reflection;
+using TaleWorlds.CampaignSystem.Actions;
+using TaleWorlds.CampaignSystem.Party;
+using TaleWorlds.CampaignSystem.Settlements;
+
+namespace GameInterface.Services.Villages.Patches
+{
+    [HarmonyPatch(typeof(ChangeVillageStateAction), "ApplyInternal")]
+    public class ChangeVillageStatePatch
+    {
+        private static AllowedInstance<Village> _allowedInstance;
+        private static readonly MethodInfo _applyInternal = typeof(ChangeVillageStateAction).GetMethod("ApplyInternal", BindingFlags.NonPublic | BindingFlags.Static);
+
+        public static bool Prefix(Village village, Village.VillageStates newState, MobileParty raiderParty)
+        {
+            if (village == _allowedInstance?.Instance) return true;
```

Comment:
We now have a method for checking if allowed

```suggestion
            if (_allowedInstance.IsAllowed(village)) return true;
```